src/ledger/grainAllocation.js

```diff
@@ -130,21 +130,16 @@ function computeImmediateReceipts(
     return [];
   }
 
-  let totalPaid = G.ZERO;
-  const receipts = mapToArray(immediateCredMap, ([address, cred]) => {
-    const amount = G.multiplyFloat(budget, cred / totalCred);
-    totalPaid = G.add(totalPaid, amount);
-    return {
-      address,
-      amount,
-    };
-  });
-  if (G.gt(totalPaid, budget)) {
-    throw new Error(
-      `invariant violation: paid ${totalPaid} greater than budget ${budget}`
-    );
-  }
-  return receipts;
+  const tuples: [NodeAddressT, number][] = mapToArray(
+    immediateCredMap,
+    (x) => x
+  );
+  const scores = tuples.map((x) => x[1]);
+  const pieces = G.splitBudget(budget, scores);
+  return tuples.map((x, i) => ({
+    address: x[0],
+    amount: pieces[i],
+  }));
```

Comment:
Nit to improve readability. Also, calling it just "tuples" is a bit vague, so even if you want to keep it would be good to have a more descriptive name (like you have for `underpaymentTuples` on line 223). 

Same with `pieces` > `grainPieces`
```suggestion
  const scores: number[] = mapToArray(
    immediateCredMap,
    (x) => x[1]
  );
  const grainPieces = G.splitBudget(budget, scores);
  return mapToArray(
    immediateCredMap,
    (x, i) => ({
      address: x[0],
      amount: grainPieces[i],
    })
  );
```